### PR TITLE
[st77xx] make reset pin as optional

### PR DESCRIPTION
**Pull request by alexandruradovici**
### Pull Request Overview

This pull request changes the st77xx driver to make the reset pin optional. Some screens' reset pin, as the one for the Pico Explorer Base (https://magpi.raspberrypi.org/articles/review-pico-explorer-base), is connected to the boards reset pin instead of a gpio pin.

### Testing Strategy

This pull request was tested using a Pico Explorer Base.

### TODO or Help Wanted

N/A

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

**Review by bradjc — APPROVED** [review]

File: boards/stm32f412gdiscovery/src/main.rs [hunk 1]
```diff
@@ -664,10 +664,12 @@ pub unsafe fn main() {
...
-            base_peripherals
-                .gpio_ports
-                .get_pin(stm32f412g::gpio::PinId::PD11)
-                .unwrap()
+            Some(
+                base_peripherals
+                    .gpio_ports
+                    .get_pin(stm32f412g::gpio::PinId::PD11)
+                    .unwrap()
+            )
```
Comment (hudson-ayers) [inline boards/stm32f412gdiscovery/src/main.rs:672]:
```suggestion
            base_peripherals
                .gpio_ports
                .get_pin(stm32f412g::gpio::PinId::PD11)
```
unless you still want to panic if `get_pin` fails

File: capsules/src/st77xx.rs [hunk 2]
```diff
@@ -238,15 +238,17 @@ impl<'a, A: Alarm<'a>, B: Bus<'a>, P: Pin> ST77XX<'a, A, B, P> {
...
         sequence_buffer: &'static mut [SendCommand],
         screen: &'static ST77XXScreen,
     ) -> ST77XX<'a, A, B, P> {
         if let Some(dc) = dc {
             dc.make_output();
         }
-        reset.make_output();
+        if let Some(reset) = reset {
+            reset.make_output();
+        }
```
Comment (hudson-ayers) [inline capsules/src/st77xx.rs:251]:
```suggestion
        reset.map(|reset| reset.make_output());
```
I consider this slightly cleaner, but its up to you

**Review by hudson-ayers — APPROVED** [review]

File: boards/stm32f412gdiscovery/src/main.rs (hunk 1 above)
Comment (alexandruradovici) [reply boards/stm32f412gdiscovery/src/main.rs:672]: I did not even think of this, thank you. :))

**Review by phil-levis — DISMISSED** [review]

File: capsules/src/st77xx.rs (hunk 2 above)
Comment (alexandruradovici) [reply capsules/src/st77xx.rs]: Good point, I updated `dc` and `reset` to map.

**Review by hudson-ayers — APPROVED** [review]

**Pull request merged**